packages/cloudflare/src/cli/templates/worker.ts

```diff
@@ -35,6 +37,17 @@ let requestHandler: NodeRequestHandler | null = null;
 export default {
   async fetch(request, env, ctx) {
     return cloudflareContextALS.run({ env, ctx, cf: request.cf }, async () => {
+      // The Middleware handler can return either a `Response` or a `Request`:
+      // - `Response`s should be returned early
+      // - `Request`s are handled by the Next server
+      const reqOrResp = await middlewareHandler(request, env);
+
+      if (reqOrResp instanceof Response) {
+        return reqOrResp;
+      }
+
+      request = reqOrResp;
```

Comment:
FYI headers from next-config and those set in the middleware (as response headers) won't work out of the box.
You'll need something like that to retrieve them https://github.com/opennextjs/opennextjs-aws/blob/bfa1a8c4056bd691fb57617dd6287693e51071b4/packages/open-next/src/core/requestHandler.ts#L54-L67 and they'll need to be merged to the response headers

Reply:
That's one of the reason why this PR doesn't target main. I'll work on that next. 
My current plan is to merge back to main when the code uses that request handler and the ON Request and Response.
After that we will need to integrate the cache.

Reply:
Ho right, i didn't see that it was not targeting main